app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedViewModel.kt

```diff
@@ -0,0 +1,171 @@
+import org.oppia.android.app.model.PromotedStory
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.domain.topic.TopicListController
+import org.oppia.android.domain.translation.TranslationController
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import org.oppia.android.util.parser.html.StoryHtmlParserEntityType
+import javax.inject.Inject
+
+/** View model for [RecentlyPlayedFragment]. */
+class RecentlyPlayedViewModel constructor(
+  private val activity: AppCompatActivity,
+  private val topicListController: TopicListController,
+  @StoryHtmlParserEntityType private val entityType: String,
+  private val resourceHandler: AppLanguageResourceHandler,
+  private val translationController: TranslationController,
+  private val promotedStoryClickListener: PromotedStoryClickListener,
+  private val internalProfileId: ProfileId,
+) {
+
+  /** Factory of RecentlyPlayedViewModel. */
+  class Factory @Inject constructor(
+    private val activity: AppCompatActivity,
+    private val topicListController: TopicListController,
+    @StoryHtmlParserEntityType private val entityType: String,
+    private val resourceHandler: AppLanguageResourceHandler,
+    private val translationController: TranslationController,
+  ) {
+
+    /** Creates an instance of [RecentlyPlayedViewModel] */
+    fun create(
+      promotedStoryClickListener: PromotedStoryClickListener,
+      internalProfileId: ProfileId
+    ): RecentlyPlayedViewModel {
+      return RecentlyPlayedViewModel(
+        activity,
+        topicListController,
+        entityType,
+        resourceHandler,
+        translationController,
+        promotedStoryClickListener,
+        internalProfileId,
+      )
+    }
+  }
+
+  /**
+   * [LiveData] with the list of recently played items for a ProfileId, organized in sections.
+   */
+  val recentlyPlayedItems: LiveData<List<RecentlyPlayedItemViewModel>> by lazy {
+    Transformations.map(promotedActivityListLiveData, ::processPromotedStoryList)
+  }
+
+  private val promotedActivityListLiveData: LiveData<PromotedActivityList> by lazy {
+    getAssumedSuccessfulPromotedActivityList()
+  }
+
+  private val promotedStoryListSummaryResultLiveData:
+    LiveData<AsyncResult<PromotedActivityList>>
+    by lazy {
+      topicListController.getPromotedActivityList(internalProfileId).toLiveData()
+    }
+
+  private fun getAssumedSuccessfulPromotedActivityList(): LiveData<PromotedActivityList> {
+    return Transformations.map(promotedStoryListSummaryResultLiveData) {
+      when (it) {
+        // If there's an error loading the data, assume the default.
+        is AsyncResult.Failure, is AsyncResult.Pending -> PromotedActivityList.getDefaultInstance()
+        is AsyncResult.Success -> it.value
+      }
+    }
+  }
+
+  private fun processPromotedStoryList(promotedActivityList: PromotedActivityList):
```

Comment:
You can break at the ``(`` instead to avoid the extra indentation being needed here (in the same way as ``addRecentlyPlayedStoryListSection`` below).

Reply:
Resolved in the latest changes.